src/common/adapters/jira.js

```diff
@@ -1,61 +1,57 @@
-import {
-  $all,
-  $find,
-  $has,
-  $text,
-  $attr,
-} from './helpers';
-
-const TYPES = ['bug', 'chore'];
-
-const normalizeType = (type) => {
-  const sanitizedType = type && type.toLowerCase();
-  if (TYPES.indexOf(sanitizedType) > -1) return sanitizedType;
-  return 'feature';
-};
-
-const ticketPageTitle = (issue) => {
-  if ($has('#summary-form', issue)) { // ticket title is currently being edited
-    return $text('#summary-form #summary', issue);
-  }
+// Jira adapter
+//
+// This adapter extracts the identifier of the selected issue from the page URL
+// and uses the Jira API to retrieve the corresponding ticket information.
+//
+// https://developer.atlassian.com/cloud/jira/software/rest/
+//
+// * Backlog and Active Sprints: https://<YOUR-SUBDOMAIN>.atlassian.net/secure/RapidBoard.jspa?…&selectedIssue=<ISSUE-KEY>
+// * Issues and filters: https://<YOUR-SUBDOMAIN>.atlassian.net/projects/<PROJECT-KEY>/issues/<ISSUE-KEY>
+// * Issue view: https://<YOUR-SUBDOMAIN>.atlassian.net/browse/<ISSUE-KEY>
+
+import client from '../client';
+
+const DOMAIN = '.atlassian.net';
+
+const issuesTabPattern = /^\/projects\/[A-Za-z0-9\-_]+\/issues\/[A-Za-z0-9\-_]+$/;
+const browseIssuePattern = /^\/browse\/[A-Za-z0-9\-_]+$/;
```

Comment:
🤔 These patterns might be a little to decipher when we're a little less restrictive:

```javascript
const issuesTabPattern = /^\/projects\/[^/]+\/issues\/[^/]+$/
```

Reply:
I'm wondering what you think about introducing a small helper, e.g. using/based on [micro-match](https://github.com/nblackburn/micro-match) or [matchit](https://github.com/lukeed/matchit) or a similar library to support more familiar URL patterns:

```javascript
import match from 'micro-match';

const { id } = match('/projects/:project/issues/:id', path);
```

For me this feels a lot more readable plus we get the values directly instead of RegExp-test first, then extract the last segment in a separate step.

Reply:
Ohh I didn't know about, I'll try it right now!